runtime/jvmti/jvmtiForceEarlyReturn.cpp

```diff
@@ -182,23 +188,23 @@ jvmtiForceEarlyReturn(jvmtiEnv* env,
 					threadToWalk = &stackThread;
 				}
 #endif /* JAVA_SPEC_VERSION >= 21 */
-				rc = findDecompileInfo(currentThread, threadToWalk, 0, &walkState);
-  				if (JVMTI_ERROR_NONE == rc) {
-  					J9Method *method = walkState.userData3;
-  					J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
-  					if (romMethod->modifiers & J9AccNative) {
-  						rc = JVMTI_ERROR_OPAQUE_FRAME;
-  					} else {
-  						J9UTF8 *sig = J9ROMMETHOD_SIGNATURE(romMethod);
-  						U_8 *data = J9UTF8_DATA(sig);
-  						U_16 length = J9UTF8_LENGTH(sig);
-  						char signatureType = data[length - 1];
-  						jvmtiParamTypes methodReturnType = JVMTI_TYPE_CVOID;
-
-  						if (('[' == data[length - 2]) || (';' == signatureType)) {
-  							signatureType = 'L';
-  						}
-  						switch(signatureType) {
+				rc = (jvmtiError)findDecompileInfo(currentThread, threadToWalk, 0, &walkState);
```

Comment:
This should include an intermediate cast through `(IDATA)` (don't change signedness and the width in a single cast).

Reply:
I thought `jvmtiError` is mapped to a `U_32` type (minimum 32 bit enum) https://github.com/eclipse-openj9/openj9/blob/master/debugtools/DDR_VM/src/com/ibm/j9ddr/StructureAliases29-edg.dat#L146, there shouldn't be any issues with direct casting from a `UDATA` to `jvmtiError`?

Reply:
That file is only relevant for DDR (and then only in IBM builds). Here, we're dealing with the C++ compiler which only sees `typedef enum jvmtiError { ... } jvmtiError;` (see [`jtmti.h.m4`](https://github.com/eclipse-openj9/openj9/blob/master/runtime/include/jvmti.h.m4#L310)). Unless otherwise specified, the underlying type for an `enum` is `int` (see https://en.cppreference.com/w/c/language/enum). So, as written, this casts `UDATA` to (effectively) `int` which is both a width change and a signedness change: the cast through `IDATA` is needed.